tests/test_frame.py

```diff
@@ -1271,6 +1271,24 @@ def test_read_excel() -> None:
         df17: dict[int | str, pd.DataFrame] = pd.read_excel("foo", sheet_name=None)
 
 
+def test_read_excel_io_types() -> None:
+    df = pd.DataFrame([[1, 2], [8, 9]], columns=["A", "B"])
+    with tempfile.NamedTemporaryFile(delete=False, suffix=".xlsx") as file:
+        as_str: str = file.name
+        df.to_excel(file.name)
+        file.close()
+
+    pd.read_excel(as_str)
+    as_path = Path(as_str)
+    pd.read_excel(as_path)
+
+    with as_path.open("rb") as as_file:
+        pd.read_excel(as_file)
+
+    as_bytes = as_path.read_bytes()
+    pd.read_excel(as_bytes)
```

Comment:
Can you indent the part that is referring to the temporary file to be within the first `with` block? 

while looking at this, it uncovered an issue with respect to us not deleting the files (which we should do), so I'll create an issue for that, but it will be easier to address if the code above is within the `with` block.

Also, we now have the convention of checking return types with new PRs, so for each `pd.read_excel()` statement, can you do something like:
```python
check(assert_type(pd.read_excel(as_bytes), pd.DataFrame), pd.DataFrame)
```

Reply:
Done.